pkg/types/aws/platform.go

```diff
@@ -10,6 +10,9 @@ type Platform struct {
 	// Region specifies the AWS region where the cluster will be created.
 	Region string `json:"region"`
 
+	// PublicZoneID specifies the ID for the public DNS Zone
+	PublicZoneID string `json:"publicZoneID"`
```

Comment:
Can we update the customization doc https://github.com/openshift/installer/blob/master/docs/user/aws/customization.md#cluster-scoped-properties

Reply:
`PublicZoneID` passed in the install-config will need validation. I think that should be addressed in this PR or broken into another card.

Reply:
I am not clear on how we should validate a zone ID. AWS only verifies if the string is not nil, and contains 1 character or more. I can certainly add a test for that, but doesn't seem sufficient.

Reply:
I've added the field to the customization doc

Reply:
we have 2 types of validations, 
pkg/types/(platform/)validations => for the structural
and pkg/asset/installconfig/platform/validations.go => for the dynamic i.e. using cloud APIs
^^ I think @patrickdillon suggestion is to add validations here to check if public zone exists and it actually matches the basedomain.

And hopefully, we can have it for all the platforms.

Reply:
I have added validations for both aws and gcp. Let me know if they look good to you ;-) I'll ping you in slack tomorrow if you haven't replied by then.